File: app/src/main/java/moe/frisk/myhooks/AppHook.java
package moe.frisk.myhooks;

import de.robv.android.xposed.callbacks.XC_LoadPackage;

public interface AppHook {

    String getKey();

    String getTitle();

    String getDescription();

    String[] getTargetPackages();

    void onPackageLoaded(XC_LoadPackage.LoadPackageParam lpparam) throws Throwable;
}

File: app/src/main/java/moe/frisk/myhooks/AppHook.kt
package moe.frisk.myhooks

import de.robv.android.xposed.callbacks.XC_LoadPackage

interface AppHook {
    val key: String
    val targetPackages: Set<String>
    fun onPackageLoaded(lpparam: XC_LoadPackage.LoadPackageParam)
}
